src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -15,12 +16,54 @@ import {
 import { Moment } from 'moment';
 
 @Injectable()
-export class CampaignActionService {
-  constructor(private store: Store<any>) {}
+export class CampaignActionService implements OnDestroy {
+  // use a Subject to filter form updates through a pipe to dispatch form changes and load preview
+  flightFormValueChanges = new Subject<{ flight: Flight; changed: boolean; valid: boolean }>();
+  flightFormValueChangesSubscription: Subscription;
+  flightFormValueChangesStream = this.flightFormValueChanges.pipe(
+    // map form field values to a proper Flight
+    map(({ flight, changed, valid }) => {
+      return {
+        flight: {
+          ...flight,
+          uncapped: flight.uncapped || false,
+          totalGoal: flight.uncapped ? 0 : +flight.totalGoal,
```

Comment:
Are these just a way to setup some default values?

Reply:
No, actually this is where it sets totalGoal and dailyMinimum to zero if the flight is uncapped. Also the incoming fields here are from the form off the valueChanges, so they are coerced to number with the `+` prefix.

Reply:
Ohhh, okay. That makes more sense.  (Also resolves some of my "where to set defaults" question).